src/mon/MgrMonitor.cc

```diff
@@ -36,6 +36,19 @@ static ostream& _prefix(std::ostream *_dout, Monitor *mon,
 		<< ").mgr e" << mgrmap.get_epoch() << " ";
 }
 
+// the system treats always_on_modules as if they provide built-in functionality
+// by ensuring that they are always enabled.
+//
+//    IMPORTANT: bump the always_on_version number when changing
+//    this list so the update is picked up correctly.
+//
+const static unsigned int always_on_version = 1;
```

Comment:
One oddity with this approach is that on an upgrade the mons go first, the mgr (probably) second. So the mons will add a new always-on module to the map before the mgr might get the latest code.  This might mean an intermittent mgr module error?

One way to fix this would be to make this a map of module to mgr version?

Reply:
I had been think we would just let mgr daemons report failed always_on modules, but for missing modules just ignore it.  That covers the upgrade case, and if we wanted to generate a warning for missing modules, we could do it on the mon side (as the mon sees the reported modules from each daemon), perhaps with a conditional that only shows the warning if the mgr's version appears to be >= the mon's version.

Reply:
Seems like we are all in agreement that the transient health checks that can appear during the upgrade transition of monitors and manager should be resolved. John's suggested solution sounds reasonable. I liked consolidating the error reporting in the manager because of the allure of easily generating very accurate error messages. But as John pointed out in another comment, errors related to running always-on modules are necessarily going to require deeper inspection if we get testing and packaging correct. So for that reason, I think splitting health check reporting responsibility is fine.

Sage suggested at last Friday's stand up that we version the always-on modules as something like:

```
always_on_modules = {
  'MIMIC': [ mimic modules ],
  'NAUTILUS': [ nautilus modules ],
}
```

The idea being that managers only look at (or are told) the lists for the versions they are running.

After staring at the ceiling for a while, I don't really see major advantages or disadvantages to either approach. John's suggestion may win a few points with being less tedious bc it retains the simpler versioning I introduced.